src/Content/CollectionView.vala

```diff
@@ -110,14 +110,18 @@ public class CollectionView : Gtk.Widget {
         listview.set_model (list_model);
 
         // Pull the posts from the list
-        shown_collection.pull_posts.begin ();
+        pull_posts ();
       } else {
         list_model = null;
         listview.set_model (null);
       }
     }
   }
 
+  internal void pull_posts() {
+    shown_collection.pull_posts.begin ();
+  }
+
```

Comment:
If I may ask, what's the reason for pulling it out of the setter?

Reply:
The `RefreshingCollectionView` calls `pull_posts` when it polls, which calls the collection method, which has a default implementation of pulling newer posts.

I could just call `shown_collection.pull_posts.begin ();` in both locations, but if there's a method then we have a clear "this is what happens when pulling new posts on a collection view".

Reply:
Looks like I've missed the second use of that method...
Then it makes sense.